docs/internal/2026-05-21-byok.md

```
@@ -0,0 +1,322 @@
+ALTER TABLE provider_key
+  ADD CONSTRAINT provider_key_token_xor
+  CHECK (token IS NULL OR token_ciphertext IS NULL);
+```
+
+If Drizzle doesn't generate this from a schema annotation, add it as a hand-written migration alongside the generated one (same migration directory, sequenced after the generated DDL).
+
+---
+
+### Ciphertext format
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)